Ionic quickstarter does not work anymore on NodeJS 10 (3.x)

Error:
```
[0;32m    amazon-ebs: + npm install[0m
[0;32m    amazon-ebs: npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.[0m
[0;32m    amazon-ebs: npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142[0m
[0;32m    amazon-ebs: npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.[0m
[0;32m    amazon-ebs: npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.[0m
[0;32m    amazon-ebs: npm WARN deprecated har-validator@5.1.5: this library is no longer supported[0m
[0;32m    amazon-ebs: npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated[0m
[0;32m    amazon-ebs: npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated[0m
[0;32m    amazon-ebs:[0m
[0;32m    amazon-ebs: > core-js@3.12.0 postinstall /tmp/workspace/unitt-cd/unitt-cd-run-unitt-fe-ionic-test-unitt-master/node_modules/core-js[0m
[0;32m    amazon-ebs: > node -e "try{require('./postinstall')}catch(e){}"[0m
[0;32m    amazon-ebs:[0m
[0;32m    amazon-ebs:[0m
[0;32m    amazon-ebs: > core-js-pure@3.13.1 postinstall /tmp/workspace/unitt-cd/unitt-cd-run-unitt-fe-ionic-test-unitt-master/node_modules/core-js-pure[0m
[0;32m    amazon-ebs: > node -e "try{require('./postinstall')}catch(e){}"[0m
[0;32m    amazon-ebs:[0m
[0;32m    amazon-ebs:[0m
[0;32m    amazon-ebs: > @angular/cli@12.0.2 postinstall /tmp/workspace/unitt-cd/unitt-cd-run-unitt-fe-ionic-test-unitt-master/node_modules/@angular/cli[0m
[0;32m    amazon-ebs: > node ./bin/postinstall/script.js[0m
[0;32m    amazon-ebs:[0m
[0;32m    amazon-ebs: npm notice created a lockfile as package-lock.json. You should commit this file.[0m
[0;32m    amazon-ebs: npm WARN notsup Unsupported engine for @angular/forms@12.0.2: wanted: {"node":"^12.14.1 || >=14.0.0"} (current: {"node":"10.24.1","npm":"6.14.12"})[0m
[0;32m    amazon-ebs: npm WARN notsup Not compatible with your version of node/npm: @angular/forms@12.0.2[0m
[0;32m    amazon-ebs: npm WARN notsup Unsupported engine for @angular/core@12.0.2: wanted: {"node":"^12.14.1 || >=14.0.0"} (current: {"node":"10.24.1","npm":"6.14.12"})[0m
[0;32m    amazon-ebs: npm WARN notsup Not compatible with your version of node/npm: @angular/core@12.0.2[0m
[0;32m    amazon-ebs: npm WARN notsup Unsupported engine for @angular/common@12.0.2: wanted: {"node":"^12.14.1 || >=14.0.0"} (current: {"node":"10.24.1","npm":"6.14.12"})[0m
[0;32m    amazon-ebs: npm WARN notsup Not compatible with your version of node/npm: @angular/common@12.0.2[0m
[0;32m    amazon-ebs: npm WARN notsup Unsupported engine for @angular/platform-browser-dynamic@12.0.2: wanted: {"node":"^12.14.1 || >=14.0.0"} (current: {"node":"10.24.1","npm":"6.14.12"})[0m
[0;32m    amazon-ebs: npm WARN notsup Not compatible with your version of node/npm: @angular/platform-browser-dynamic@12.0.2[0m
[0;32m    amazon-ebs: npm WARN notsup Unsupported engine for @angular/router@12.0.2: wanted: {"node":"^12.14.1 || >=14.0.0"} (current: {"node":"10.24.1","npm":"6.14.12"})[0m
[0;32m    amazon-ebs: npm WARN notsup Not compatible with your version of node/npm: @angular/router@12.0.2[0m
[0;32m    amazon-ebs: npm WARN notsup Unsupported engine for @angular/platform-browser@12.0.2: wanted: {"node":"^12.14.1 || >=14.0.0"} (current: {"node":"10.24.1","npm":"6.14.12"})[0m
[0;32m    amazon-ebs: npm WARN notsup Not compatible with your version of node/npm: @angular/platform-browser@12.0.2[0m
[0;32m    amazon-ebs: npm WARN notsup Unsupported engine for @stencil/core@2.5.2: wanted: {"node":">=12.10.0","npm":">=6.0.0"} (current: {"node":"10.24.1","npm":"6.14.12"})[0m
[0;32m    amazon-ebs: npm WARN notsup Not compatible with your version of node/npm: @stencil/core@2.5.2[0m
[0;32m    amazon-ebs: npm WARN notsup Unsupported engine for @angular-devkit/build-angular@12.0.2: wanted: {"node":"^12.14.1 || ^14.0.0","npm":"^6.11.0 || ^7.5.6","yarn":">= 1.13.0"} (current: {"node":"10.24.1","npm":"6.14.12"})[0m
[0;32m    amazon-ebs: npm WARN notsup Not compatible with your version of node/npm: @angular-devkit/build-angular@12.0.2[0m
[0;32m    amazon-ebs: npm WARN notsup Unsupported engine for @angular/cli@12.0.2: wanted: {"node":"^12.14.1 || ^14.0.0","npm":"^6.11.0 || ^7.5.6","yarn":">= 1.13.0"} (current: {"node":"10.24.1","npm":"6.14.12"})[0m
[0;32m    amazon-ebs: npm WARN notsup Not compatible with your version of node/npm: @angular/cli@12.0.2[0m
[0;32m    amazon-ebs: npm WARN notsup Unsupported engine for @angular/compiler-cli@12.0.2: wanted: {"node":"^12.14.1 || >=14.0.0"} (current: {"node":"10.24.1","npm":"6.14.12"})[0m
[0;32m    amazon-ebs: npm WARN notsup Not compatible with your version of node/npm: @angular/compiler-cli@12.0.2[0m
[0;32m    amazon-ebs: npm WARN notsup Unsupported engine for @angular/language-service@12.0.2: wanted: {"node":"^12.14.1 || >=14.0.0"} (current: {"node":"10.24.1","npm":"6.14.12"})[0m
[0;32m    amazon-ebs: npm WARN notsup Not compatible with your version of node/npm: @angular/language-service@12.0.2[0m
[0;32m    amazon-ebs: npm WARN notsup Unsupported engine for @angular/compiler@12.0.2: wanted: {"node":"^12.14.1 || >=14.0.0"} (current: {"node":"10.24.1","npm":"6.14.12"})[0m
[0;32m    amazon-ebs: npm WARN notsup Not compatible with your version of node/npm: @angular/compiler@12.0.2[0m
[0;32m    amazon-ebs: npm WARN notsup Unsupported engine for @angular-devkit/architect@0.1200.2: wanted: {"node":"^12.14.1 || ^14.0.0","npm":"^6.11.0 || ^7.5.6","yarn":">= 1.13.0"} (current: {"node":"10.24.1","npm":"6.14.12"})[0m
[0;32m    amazon-ebs: npm WARN notsup Not compatible with your version of node/npm: @angular-devkit/architect@0.1200.2[0m
[0;32m    amazon-ebs: npm WARN notsup Unsupported engine for @angular-devkit/core@12.0.2: wanted: {"node":"^12.14.1 || ^14.0.0","npm":"^6.11.0 || ^7.5.6","yarn":">= 1.13.0"} (current: {"node":"10.24.1","npm":"6.14.12"})[0m
[0;32m    amazon-ebs: npm WARN notsup Not compatible with your version of node/npm: @angular-devkit/core@12.0.2[0m
[0;32m    amazon-ebs: npm WARN notsup Unsupported engine for @angular-devkit/build-optimizer@0.1200.2: wanted: {"node":"^12.14.1 || ^14.0.0","npm":"^6.11.0 || ^7.5.6","yarn":">= 1.13.0"} (current: {"node":"10.24.1","npm":"6.14.12"})[0m
[0;32m    amazon-ebs: npm WARN notsup Not compatible with your version of node/npm: @angular-devkit/build-optimizer@0.1200.2[0m
[0;32m    amazon-ebs: npm WARN notsup Unsupported engine for @angular-devkit/build-webpack@0.1200.2: wanted: {"node":"^12.14.1 || ^14.0.0","npm":"^6.11.0 || ^7.5.6","yarn":">= 1.13.0"} (current: {"node":"10.24.1","npm":"6.14.12"})[0m
[0;32m    amazon-ebs: npm WARN notsup Not compatible with your version of node/npm: @angular-devkit/build-webpack@0.1200.2[0m
[0;32m    amazon-ebs: npm WARN notsup Unsupported engine for @ngtools/webpack@12.0.2: wanted: {"node":"^12.14.1 || ^14.0.0","npm":"^6.11.0 || ^7.5.6","yarn":">= 1.13.0"} (current: {"node":"10.24.1","npm":"6.14.12"})[0m
[0;32m    amazon-ebs: npm WARN notsup Not compatible with your version of node/npm: @ngtools/webpack@12.0.2[0m
[0;32m    amazon-ebs: npm WARN notsup Unsupported engine for css-minimizer-webpack-plugin@3.0.0: wanted: {"node":">= 12.13.0"} (current: {"node":"10.24.1","npm":"6.14.12"})[0m
[0;32m    amazon-ebs: npm WARN notsup Not compatible with your version of node/npm: css-minimizer-webpack-plugin@3.0.0[0m
[0;32m    amazon-ebs: npm WARN notsup Unsupported engine for open@8.0.2: wanted: {"node":">=12"} (current: {"node":"10.24.1","npm":"6.14.12"})[0m
[0;32m    amazon-ebs: npm WARN notsup Not compatible with your version of node/npm: open@8.0.2[0m
[0;32m    amazon-ebs: npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@~2.3.1 (node_modules/chokidar/node_modules/fsevents):[0m
[0;32m    amazon-ebs: npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})[0m
[0;32m    amazon-ebs: npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.2.7 (node_modules/webpack-dev-server/node_modules/chokidar/node_modules/fsevents):[0m
[0;32m    amazon-ebs: npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})[0m
[0;32m    amazon-ebs: npm WARN notsup Unsupported engine for @angular-devkit/schematics@12.0.2: wanted: {"node":"^12.14.1 || ^14.0.0","npm":"^6.11.0 || ^7.5.6","yarn":">= 1.13.0"} (current: {"node":"10.24.1","npm":"6.14.12"})[0m
[0;32m    amazon-ebs: npm WARN notsup Not compatible with your version of node/npm: @angular-devkit/schematics@12.0.2[0m
[0;32m    amazon-ebs: npm WARN notsup Unsupported engine for @schematics/angular@12.0.2: wanted: {"node":"^12.14.1 || ^14.0.0","npm":"^6.11.0 || ^7.5.6","yarn":">= 1.13.0"} (current: {"node":"10.24.1","npm":"6.14.12"})[0m
[0;32m    amazon-ebs: npm WARN notsup Not compatible with your version of node/npm: @schematics/angular@12.0.2[0m
[0;32m    amazon-ebs: npm WARN ajv-keywords@3.5.2 requires a peer of ajv@^6.9.1 but none is installed. You must install peer dependencies yourself.[0m
[0;32m    amazon-ebs:[0m
[0;32m    amazon-ebs: added 1584 packages from 1342 contributors in 110.678s[0m
[0;32m    amazon-ebs:[0m
[0;32m    amazon-ebs: 128 packages are looking for funding[0m
[0;32m    amazon-ebs:   run `npm fund` for details[0m
[0;32m    amazon-ebs:[0m
[0;32m    amazon-ebs: [Pipeline] sh[0m
[0;32m    amazon-ebs: + ionic build --prod[0m
[0;32m    amazon-ebs: > ng run app:build:production[0m
[0;32m    amazon-ebs: Node.js version v10.24.1 detected.[0m
[0;32m    amazon-ebs: The Angular CLI requires a minimum Node.js version of either v12.14 or v14.15.[0m
[0;32m    amazon-ebs:[0m
[0;32m    amazon-ebs: Please update your Node.js version or visit https://nodejs.org/ for additional instructions.[0m
[0;32m    amazon-ebs:[0m
[0;32m    amazon-ebs: [ERROR] An error occurred while running subprocess ng.[0m
[0;32m    amazon-ebs:[0m
[0;32m    amazon-ebs:         ng run app:build:production exited with exit code 3.[0m
[0;32m    amazon-ebs:[0m
[0;32m    amazon-ebs:         Re-running this command with the --verbose flag may provide more information.[0m
[0;32m    amazon-ebs: [Pipeline] }[0m
[0;32m    amazon-ebs: [Pipeline] // withEnv[0m
[0;32m    amazon-ebs: [Pipeline] }[0m
[0;32m    amazon-ebs: [Pipeline] // stage[0m
[0;32m    amazon-ebs: [Pipeline] }[0m
[0;32m    amazon-ebs: [Pipeline] // wrap[0m
[0;32m    amazon-ebs: [Pipeline] stage[0m
[0;32m    amazon-ebs: [Pipeline] { (odsPipeline error)[0m
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,8 @@
 # Changelog
 
+### Fixed
+- Fix Ionic quickstarter build after provisioning ([#595](https://github.com/opendevstack/ods-quickstarters/issues/595))
+
 ## Unreleased
 
 - ds-rshiny cleanup cloudera dependency ([#540](https://github.com/opendevstack/ods-quickstarters/pull/540))

--- a/fe-ionic/Jenkinsfile.template
+++ b/fe-ionic/Jenkinsfile.template
@@ -22,7 +22,7 @@ odsComponentPipeline(
 def stageBuild(def context) {
   stage('Build') {
     withEnv(["TAGVERSION=${context.tagversion}", "NEXUS_HOST=${context.nexusHost}"]) {
-      sh "npm i -g @ionic/cli@6.3.0"
+      sh "npm i -g @ionic@5.4.16"
       sh "npm install"
       if ('master'.equals(context.gitBranch)) {
         sh "ionic build --prod"

